ada_feeding/ada_feeding/helpers.py

```diff
@@ -92,6 +93,87 @@ def quat_between_vectors(vec_from: Vector3, vec_to: Vector3) -> Quaternion:
     return ret
 
 
+def add_update_static_tf(
```

Comment:
Do you ever intend to let a behavior get the transforms? (e.g., to get the food frame?) If so, I think you should make an analogous getter function for this, as opposed to letting behaviors do it themselves. (To ensure locking logic and such is handled correctly)

Reply:
Nope, behaviors should get transforms through the TransformListener (`get_tf_object()` later in the file).

Reply:
Got it. In that case, why do you need to store a dict of the transforms on the blackboard? If you just publish transforms directly to the Broadcaster as this function gets called, shouldn't the Broadcaster handle overriding old transforms? I'm not sure if the dict functionality is necessary.

Reply:
We do need the dict.

The STB handles the latching (i.e. holding on to the last published messages), but `sendTransform` just forwards the transform / list of transforms to the publisher.

Actually handling adding transforms to a list is a feature [present in CPP](https://github.com/ros2/geometry2/blob/a73160df358c52d1f506fce21f15f8243e9b87f4/tf2_ros/src/static_transform_broadcaster.cpp#L52) but [not Python](https://github.com/ros2/geometry2/blob/humble/tf2_ros_py/tf2_ros/static_transform_broadcaster.py#L64)

I'll make an issue upstream

Reply:
See https://github.com/ros2/geometry2/issues/631